### ISSUE TITLE
Executable bit not preserved/set for universal Mach-O binaries in `.app` tree after a Velopack update

### ISSUE THREAD

**Issue by sharadhr** [velopack/velopack#126]
Velopack is supposed to set the executable bit for all Mach-O binaries after applying an update, but this doesn't happen for universal Mach-O binaries, resulting in a possibly broken state after the update. An example from [the Discord server](https://discord.com/channels/767856501477343282/947444323765583913/1246154799641268274):

```
$ ls -lah
total 123464
drwxr-xr-x   4 user  wheel   128B 31 May 18:33 .
drwxr-xr-x  86 user  wheel   2.7K 31 May 18:31 ..
-rwxr-xr-x@  1 user  wheel    20M 31 May 18:31 bin_arm64
-rw-r--r--@  1 user  wheel    40M 31 May 18:31 bin_universal
$ file bin_arm64
bin_arm64: Mach-O 64-bit executable arm64
$ file bin_universal
bin_universal: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
bin_universal (for architecture x86_64):    Mach-O 64-bit executable x86_64
bin_universal (for architecture arm64):    Mach-O 64-bit executable arm64
```

**Comment by ILW8** [comment]
It looks like universal mach-o handling isn't present. Adding support seems a bit hacky at best, but at least there is Apple documentation on how to handle the magic collision they've brought upon themselves.

```diff
diff --git a/src/Velopack.Packaging.Unix/MachO.cs b/src/Velopack.Packaging.Unix/MachO.cs
index c6bf4a5f..c56c8b44 100644
--- a/src/Velopack.Packaging.Unix/MachO.cs
+++ b/src/Velopack.Packaging.Unix/MachO.cs
@@ -7,7 +7,9 @@ private enum MagicMachO : uint
         MH_MAGIC = 0xfeedface,
         MH_CIGAM = 0xcefaedfe,
         MH_MAGIC_64 = 0xfeedfacf,
-        MH_CIGAM_64 = 0xcffaedfe
+        MH_CIGAM_64 = 0xcffaedfe,
+        FAT_MAGIC = 0xcafebabe, // https://developer.apple.com/documentation/kernel/fat_header/1558632-magic/
+        FAT_CIGAM = 0xbebafeca
     }
 
     public static bool IsMachOImage(string filePath)
@@ -17,7 +19,17 @@ public static bool IsMachOImage(string filePath)
                 return false;
 
             uint magic = reader.ReadUInt32();
-            return Enum.IsDefined(typeof(MagicMachO), magic);
+            if (!Enum.IsDefined(typeof(MagicMachO), magic)) {
+                return false;
+            }
+
+            if ((MagicMachO) magic is not (MagicMachO.FAT_CIGAM or MagicMachO.FAT_MAGIC)) {
+                return true;
+            }
+
+            // https://opensource.apple.com/source/file/file-80.40.2/file/magic/Magdir/cafebabe.auto.html
+            uint archNumber = reader.ReadUInt32();
+            return archNumber < 0x20;
         }
     }
 }
```

**Comment by caesay** [comment]
Fixed in 657503144239f0488a16d825b0dcc4abda8f6720 and released in 0.0.517
